src/DependencyInjection/Configuration.php

```diff
@@ -18,9 +18,11 @@ class Configuration implements ConfigurationInterface
     public function getConfigTreeBuilder(): TreeBuilder
     {
         $treeBuilder = new TreeBuilder('knpu_oauth2_client');
-        $rootNode = method_exists($treeBuilder, 'getRootNode')
-            ? $treeBuilder->getRootNode()
-            : $treeBuilder->root('knpu_oauth2_client');
+        if (method_exists($treeBuilder, 'root')) {
+            $rootNode = $treeBuilder->root('knpu_oauth2_client'); // BC for Symfony < 4.2
+        } else {
+            $rootNode = $treeBuilder->getRootNode();
+        }
```

Comment:
What caused this change? I think it's not correct. The `root()` method is deprecated, but (for example) it exists in Symfony 4.3. So, this would cause a user in 4.3/4.4 to use the `root()` method and see a deprecation warning. We need to check for the presence of `getRootNode()`, which is the newer method.

Reply:
I was trying to avoid adding more PHPStan `ignoreErrors`, but since that call must be there for <4.2, I think that it is inevitable for now.